Frends.Community.Email/EmailTask.cs

```diff
@@ -183,95 +184,112 @@ public static async Task<Output> SendEmailToExchangeServer([PropertyTab] Exchang
+            var allAttachmentFilePaths = new List<string>();
+
             foreach (var attachment in attachments)
             {
-                if (attachment.AttachmentType == AttachmentType.FileAttachment)
-                {
-                    var allAttachmentFilePaths = GetAttachmentFiles(attachment.FilePath);
-
-                    if (attachment.ThrowExceptionIfAttachmentNotFound && allAttachmentFilePaths.Count == 0) throw new FileNotFoundException(string.Format("The given filepath \"{0}\" had no matching files", attachment.FilePath), attachment.FilePath);
 
-                    if (allAttachmentFilePaths.Count == 0 && !attachment.SendIfNoAttachmentsFound)
-                    {
-                        var emailSendFailed = new MessageAttachmentsCollectionPage
-                        {
-                            new FileAttachment { Name = attachment.FilePath, ContentType = "FailedEmail" }
-                        };
-                        return emailSendFailed;
-                    }
-
-                    foreach (var filePath in allAttachmentFilePaths)
+                string tempFilePath = "";
+                if (attachment.AttachmentType == AttachmentType.AttachmentFromString)
+                {
+                    // Create attachment only if content is not empty.
+                    if (!string.IsNullOrEmpty(attachment.StringAttachment.FileContent))
                     {
-                        cancellationToken.ThrowIfCancellationRequested();
-                        var attachmentContent = File.ReadAllBytes(filePath);
+                        tempFilePath = CreateTemporaryFile(attachment);
+                        var attachmentContent = File.ReadAllBytes(tempFilePath);
                         var oneAttachment = new FileAttachment
                         {
                             ODataType = "#microsoft.graph.fileAttachment",
                             ContentBytes = attachmentContent,
-                            ContentType = MimeTypes.GetMimeType(filePath),
-                            ContentId = "Something",
-                            Name = Path.GetFileName(filePath)
+                            ContentType = MimeTypes.GetMimeType(tempFilePath),
+                            Name = Path.GetFileName(tempFilePath)
                         };
-                        attachmentList.Add(oneAttachment);
+                        allAttachmentFilePaths.Add(tempFilePath);
                     }
                 }
+                else
+                    allAttachmentFilePaths = GetAttachmentFiles(attachment.FilePath);
 
-                if (attachment.AttachmentType == AttachmentType.AttachmentFromString)
+                if (attachment.ThrowExceptionIfAttachmentNotFound && allAttachmentFilePaths.Count == 0) throw new FileNotFoundException(string.Format("The given filepath \"{0}\" had no matching files", attachment.FilePath), attachment.FilePath);
```

Comment:
Please don't use string.Format when there's much more elegant way of building a string $"{attachment.FilePath}"